src/rules/no-stale-state-across-await.ts

```
@@ -0,0 +1,182 @@
+        // Walk the function body
+        if (node.body) {
+          walkNode(node.body);
+        }
+
+        // Check for violations
+        const setterCallsByName = new Map<string, { name: string; position: number }[]>();
+        for (const call of setterCalls) {
+          if (!setterCallsByName.has(call.name)) {
+            setterCallsByName.set(call.name, []);
+          }
+          setterCallsByName.get(call.name)!.push(call);
```

Biome (1.9.4)

[error] Forbidden non-null assertion.

Unsafe fix: Replace with optional chain operator ?. This operator includes runtime checks, so it is safer than the compile-only non-null assertion operator


(lint/style/noNonNullAssertion)